src/applications/caregivers/utils/helpers.js

```diff
@@ -1,150 +1,112 @@
-        }),
-      {},
-    );
-
-    return Object.assign(labels, stateLabels);
-  },
-  {},
-);
-
-// Turns the facility list for each state into an array of strings
-export const medicalCentersByState = mapValues(caregiverFacilities, val =>
-  val.map(center => center.code),
-);
-
-// transforms forData to match fullSchema structure for backend submission
-export const submitTransform = (formConfig, form) => {
-  const hasPrimary = form.data['view:hasPrimaryCaregiver'] ? 'primary' : null;
 
-  const hasSecondaryOne = form.data['view:hasSecondaryCaregiverOne']
+// transforms form data to match fullSchema structure for backend submission
+export const submitTransform = (formConfig, form) => {
+  const { data: formData } = form;
+  const hasPrimary = formData['view:hasPrimaryCaregiver'] ? 'primary' : null;
+  const hasSecondaryOne = formData['view:hasSecondaryCaregiverOne']
     ? 'secondaryOne'
     : null;
-
-  const hasSecondaryTwo = form.data['view:hasSecondaryCaregiverTwo']
+  const hasSecondaryTwo = formData['view:hasSecondaryCaregiverTwo']
     ? 'secondaryTwo'
     : null;
 
   // creates chapter objects by matching chapter prefixes
-  const buildChapterSortedObject = (data, dataPrefix) => {
+  const buildChapterSortedObject = dataPrefix => {
     // check to make sure there is a keyName
     if (dataPrefix === null) return {};
-    const keys = Object.keys(data);
 
     // matches prefix to fullSchema chapter object labels/keys
     const getChapterName = key => {
-      switch (key) {
-        case 'veteran':
-          return 'veteran';
-        case 'primary':
-          return 'primaryCaregiver';
-        case 'secondaryOne':
-          return 'secondaryCaregiverOne';
-        case 'secondaryTwo':
-          return 'secondaryCaregiverTwo';
-        default:
-          return null;
-      }
+      const keyMap = {
+        veteran: 'veteran',
+        primary: 'primaryCaregiver',
+        secondaryOne: 'secondaryCaregiverOne',
+        secondaryTwo: 'secondaryCaregiverTwo',
+      };
+      return keyMap[key] || null;
     };
 
+    const dataKeys = Object.keys(formData).filter(k => k.includes(dataPrefix));
     const chapterName = getChapterName(dataPrefix);
-
-    const sortedDataByChapter = {
-      [chapterName]: {},
-      signAsRepresentative: false, // sign as veteran is default and false
-    };
-
-    const lowerCaseFirstLetter = string =>
-      string.charAt(0).toLowerCase() + string.slice(1);
-
-    // maps over all keys, and creates objects of the same prefix then removes prefix
-    keys.map(key => {
-      if (key === 'signAsRepresentativeDocumentUpload') {
-        /* if user submits a document via upload page, add the guid to the formData
-          otherwise delete object and move on to next keys */
-        if (isEmpty(data[key]) || data.signAsRepresentativeYesNo !== 'yes') {
-          return delete sortedDataByChapter.poaAttachmentId;
-        }
-
-        const documentUpload = data[key][0].guid;
```

Comment:
Since we're not doing a ridiculous map over all keys, the sign as representative logic has been moved to its own method.